### ISSUE TITLE
Make it possible to disallow schema lookups on disk

### ISSUE THREAD

**Issue by johanfylling** [open-policy-agent/opa#5676]
Currently, it is possible to use the `file://` scheme in `$ref` schema references to lookup schemas on the local file system for type-checking: 

```rego
package test

# METADATA
# schemas:
# - input: {"$ref": "file://schema.json"}
p {
    input == 42
}
```

Out of concern for security, unnecessary/unwanted file reads should be avoided; so by default, `$ref`s with the `file://` scheme should not be allowed. File-reads could be allowed through a `capabilities` attribute (but might be an unnecessary addition).

**Comment by ashutosh-narkar** [comment]
> File-reads could be allowed through a capabilities attribute (but might be an unnecessary addition).

Are you suggesting extending the `capabilities` support to specify file refs?

**Comment by johanfylling** [comment]
Yes @ashutosh-narkar. I'm thinking a boolean attribute to allow all or no file refs might be enough, though.

**Comment by ashutosh-narkar** [comment]
Sounds good. I think it would be better if we specify the refs in the new field instead of a boolean attribute imo.

**Comment by ashutosh-narkar** [comment]
> so by default, $refs with the file:// scheme should not be allowed

@johan could this be part of strict type checking?

**Comment by johanfylling** [comment]
I think file $refs should probably be disabled by default, and something you need to opt into. 
But I also don't think the security concern is very big here.

**Comment by srenatus** [comment]
Conceptually, this would fit nicely with capabilities -- there's a capability for network access, and there could be another one for file system access. After all, it's been modeled after deno's capabilities, which have the same distinction. But I agree, that this might not be necessary from a security perspective. **Unless** you get an error of some sort that contains the schema _including bits read from disk_. We'd have to check if that's possible.
